c7n/resources/aws.py

```diff
@@ -624,7 +624,8 @@ def initialize_policies(self, policy_collection, options):
                 candidate = candidates and candidates[0] or 'us-east-1'
                 svc_regions = [candidate]
             elif 'all' in options.regions:
-                svc_regions = list(set(available_regions).intersection(enabled_regions))
+                svc_regions = list(set(available_regions).intersection(enabled_regions) -
```

Comment:
so given -r us-east-1 -r us-east-2 -r us-west-2. --exclude-region us-east-1

this would still run in us-east-1

Reply:
I originally had it outside the `if` so that it would always be applied, regardless of the value for `--region`.  I wasn't sure if it was better to have the option only work with `all` or for it to work all the time.  Do you prefer it to work all the time?

Reply:
i think it probably make sense to work in both situations, ie explicit deny trumps explicit allow

Reply:
The global resources use `us-east-1` according to the previous comment and lines in the code:
```
            # its a global service/endpoint, use user provided region
            # or us-east-1.
            if not available_regions and options.regions:
                candidates = [r for r in options.regions if r != 'all']
                candidate = candidates and candidates[0] or 'us-east-1'
```
If `us-east-1` is excluded here, should it skip global resources?

Reply:
thats a good question, thats a bit non obvious, but yes, its probably the right thing to do re skip. its common to want to do the exclude based on network connectivity to a region, or region locality. we probably could use some docs in all the global  resources re their pointing to us-east-1 (although a few like global accelerator do other regions).